pkg/controller/controller.go

```diff
@@ -1304,9 +1305,20 @@ func updatedStatefulSet(
 	expected.Status = actual.DeepCopy().Status
 
 	copyAnnotations(expected, actual)
+	updateSpecTemplate(expected)
 	return expected, true, nil
 }
 
+// updates spec template so that rollout update is always performed.
+func updateSpecTemplate(sts *appsv1.StatefulSet) {
+	if sts.Spec.Template.Annotations == nil {
+		sts.Spec.Template.Annotations = map[string]string{}
+	}
+
+	now, _ := metav1.Now().MarshalQueryParameter()
+	sts.Spec.Template.Annotations[annotations.Rollout] = now
```

Comment:
I wonder if we should only do this if the user opts in to this behavior by setting a corresponding field in the cluster spec? I just worry it might come as a surprise to someone that was expecting the old behavior of the operator not performing an update if the statefulsets hadn't changed.

Reply:
You still need to update sts with `update` annotation for rollout to happen, I am assuming that if someone already added this annotation they probably expect that sts should be updated. Otherwise what's the point for adding an `update` annotation in the first place?

Reply:
The main time I can think of that having the operator skip statefulsets that don't need to be updated has come in handy is when an update to a cluster has been interrupted partway through. For example, we would set the annotation on all statefulsets, but then if we ran into an issue we could remove the annotation from all statefulsets to ensure the operator doesn't perform any more updates while we debug any issue that we encountered. Once we've decided to continue with the update we could once again set the annotation on all statefulsets and leave it up to the operator to determine which statefulsets still need to be updated and which don't, skipping any nodes which have already been updated. It's admittedly not a very common use case but I worry that if anyone has become used to it then they may be surprised if the behavior changes here. But I'm also not convinced that we necessarily need to keep supporting this behavior. @schallert what do you think?